webapi/src/main/scala/org/knora/webapi/util/StringFormatter.scala

```diff
@@ -699,6 +700,9 @@ class StringFormatter private(val knoraApiHostAndPort: Option[String]) {
     private val SalsahGuiAttributeRegex: Regex =
         """^(\p{L}+)=(.+)$""".r
 
+    // A regex for matching a string containing an email address.
+    private val EmailAddressRegex: Regex = ("""^([0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*@(([0-9a-zA-Z])+([-\w]*[0-9a-zA-Z])*\.)+[a-zA-Z]{2,9})$""").r
```

Comment:
This won't work for email address with non-ASCII characters, such as email addresses at [Internationalized Domain Names](https://www.icann.org/resources/pages/idn-2012-02-25-en).  Some [examples](https://en.wikipedia.org/wiki/Email_address#Internationalization_examples):

* Latin alphabet with diacritics: Pelé@example.com
* Greek alphabet: δοκιμή@παράδειγμα.δοκιμή
* Traditional Chinese characters: 我買@屋企.香港
* Japanese characters: 二ノ宮@黒川.日本
* Cyrillic characters: чебурашка@ящик-с-апельсинами.рф
* Devanagari characters: संपर्क@डाटामेल.भारत

There seems to be a lot of debate on StackOverflow and elsewhere about whether it's really possible to validate an email address with a regex. Other options:

* An Apache Commons class called [EmailValidator](https://commons.apache.org/proper/commons-validator/apidocs/index.html?org/apache/commons/validator/routines/EmailValidator.html). I've just tried it, and it rejects the Greek and Japanese examples above.
* [The javax.mail API](https://kodejava.org/how-do-i-validate-email-address-using-java-mail-api/), which you can [get from Maven](https://mvnrepository.com/artifact/com.sun.mail/javax.mail/1.6.2). But it rejects the Devanagari example above, which is actually the real contact email address of [this hosting service in India](http://www.xn--c2bd4bq1db8d.xn--h2brj9c/).

My suggestion: use a very minimal regex like `^.*@.*$`, and actually send an email confirmation to the user to check that they received the email. Because even if the address is syntactically valid, the user could still have mistyped it. If they can actually receive a message there, we know it's OK.

Reply:
Yes, the email confirmation is a good suggestion. I've opened an issue for this #1065.

Reply:
I guess we can require at least one character before and after the `@`. :)